### ISSUE TITLE
Build issues with Autoconf 2.71

### ISSUE THREAD

**Issue by eggert** [gwsw/less#320]
Fedora 37, which uses Autoconf 2.71, complains mightily about `less` master; see below. I'll see if I come up with fixes. Fixes will likely assume Autoconf 2.53 (2002) or later, but that's safe now.

```
$ make -f Makefile.aut
mv -f ./help.c ./help.c.old
mv: cannot stat './help.c': No such file or directory
make: [Makefile.aut:59: help.c] Error 1 (ignored)
rm -rf help.c
./mkhelp.pl < less.hlp > help.c
if cmp -s help.c help.c.old; then mv -f help.c.old help.c; fi
mv -f ./funcs.h ./funcs.h.old
mv: cannot stat './funcs.h': No such file or directory
make: [Makefile.aut:68: funcs.h] Error 1 (ignored)
grep -h '^public [^;]*$' ./main.c ./screen.c ./brac.c ./ch.c ./charset.c ./cmdbuf.c ./command.c ./cvt.c ./decode.c ./edit.c ./filename.c ./forwback.c ./help.c ./ifile.c ./input.c ./jump.c ./line.c ./linenum.c ./lsystem.c ./mark.c ./optfunc.c ./option.c ./opttbl.c ./os.c ./output.c ./pattern.c ./position.c ./prompt.c ./search.c ./signal.c ./tags.c ./ttyin.c ./version.c ./xbuf.c | sed 's/$/;/' >./funcs.h
./mkutable -f2 Mn Me -- unicode/UnicodeData.txt > compose.uni
./mkutable -f2 Cf -- unicode/UnicodeData.txt > fmt.uni
./mkutable -f2 Cc Cs Co Zl Zp -- unicode/UnicodeData.txt > ubin.uni
cd .; autoheader; autoconf
configure.ac:14: warning: The macro `AC_CONFIG_HEADER' is obsolete.
configure.ac:14: You should run autoupdate.
./lib/autoconf/status.m4:719: AC_CONFIG_HEADER is expanded from...
configure.ac:14: the top level
configure.ac:18: warning: The macro `AC_ISC_POSIX' is obsolete.
configure.ac:18: You should run autoupdate.
./lib/autoconf/specific.m4:550: AC_ISC_POSIX is expanded from...
configure.ac:18: the top level
configure.ac:39: warning: The macro `AC_HEADER_STDC' is obsolete.
configure.ac:39: You should run autoupdate.
./lib/autoconf/headers.m4:704: AC_HEADER_STDC is expanded from...
configure.ac:39: the top level
configure.ac:47: warning: The macro `AC_HEADER_TIME' is obsolete.
configure.ac:47: You should run autoupdate.
./lib/autoconf/headers.m4:743: AC_HEADER_TIME is expanded from...
configure.ac:47: the top level
configure.ac:74: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:74: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:74: the top level
configure.ac:87: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:87: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:87: the top level
configure.ac:100: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:100: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:100: the top level
configure.ac:113: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:113: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:113: the top level
configure.ac:126: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:126: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:126: the top level
configure.ac:139: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:139: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:139: the top level
configure.ac:153: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:153: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:153: the top level
configure.ac:168: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:168: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:168: the top level
configure.ac:181: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:181: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:181: the top level
configure.ac:260: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:260: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
configure.ac:260: the top level
configure.ac:263: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:263: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
configure.ac:263: the top level
configure.ac:266: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:266: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
configure.ac:266: the top level
configure.ac:269: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:269: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
configure.ac:269: the top level
configure.ac:274: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:274: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
configure.ac:274: the top level
configure.ac:283: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:283: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
configure.ac:283: the top level
configure.ac:287: warning: The macro `AC_TYPE_SIGNAL' is obsolete.
configure.ac:287: You should run autoupdate.
./lib/autoconf/types.m4:776: AC_TYPE_SIGNAL is expanded from...
configure.ac:287: the top level
configure.ac:292: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:292: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:292: the top level
configure.ac:299: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:299: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:299: the top level
configure.ac:306: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:306: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:306: the top level
configure.ac:316: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:316: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:316: the top level
configure.ac:323: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:323: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:323: the top level
configure.ac:336: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:336: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:336: the top level
configure.ac:342: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:342: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:342: the top level
configure.ac:349: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:349: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:349: the top level
configure.ac:355: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:355: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:355: the top level
configure.ac:364: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:364: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:364: the top level
configure.ac:370: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:370: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:370: the top level
configure.ac:377: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:377: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:377: the top level
configure.ac:383: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:383: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:383: the top level
configure.ac:393: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:393: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:393: the top level
configure.ac:411: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:411: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:411: the top level
configure.ac:433: warning: The macro `AC_TRY_RUN' is obsolete.
configure.ac:433: You should run autoupdate.
./lib/autoconf/general.m4:2997: AC_TRY_RUN is expanded from...
configure.ac:433: the top level
configure.ac:451: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:451: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:451: the top level
configure.ac:494: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:494: You should run autoupdate.
./lib/autoconf/general.m4:2920: AC_TRY_LINK is expanded from...
configure.ac:494: the top level
$ autoupdate
configure.ac:39: warning: The preprocessor macro `STDC_HEADERS' is obsolete.
  Except in unusual embedded environments, you can safely include all
  ISO C90 headers unconditionally.
configure.ac:47: warning: Update your code to rely only on HAVE_SYS_TIME_H,
then remove this warning and the obsolete code below it.
All current systems provide time.h; it need not be checked for.
Not all systems provide sys/time.h, but those that do, all allow
you to include it and time.h simultaneously.
configure.ac:287: warning: your code may safely assume C89 semantics that RETSIGTYPE is void.
Remove this warning and the `AC_CACHE_CHECK' when you adjust the code.
```

**Comment by eggert** [comment]
I've created PR #321 to try to fix this.

**Comment by gwsw** [comment]
@eggert Hi Paul,
In cfde4d318d521f8789cc5163cf1c62e488477249 I've updated via autoupdate 2.69, which is the newest version for my fedora system. I chose to retain RETSIGTYPE as a manually defined symbol. 

It looks like you may still get the STDC_HEADER warning with autoconf 2.71. Let me know if you see any other messages.

**Comment by gwsw** [comment]
Sorry, wrong commit id, it should be 3239d7382df5aa3c0231ae0488169d3c8fe3158a.

**Comment by eggert** [comment]
> It looks like you may still get the STDC_HEADER warning with autoconf 2.71. Let me know if you see any other messages.

Unfortunately I still get complaints about `AC_CONFIG_HEADER` and `AC_HEADER_TIME`. All the complaints (including the one about `AC_HEADER_STDC` that you mentioned) are easy to fix and the fixes make `configure` smaller and faster, so I created PR #324 to address this. That PR has an additional patch to make `configure` even smaller and less chatty, without changing behavior. (You can close PR #320 as it's now superseded.)

**Comment by gwsw** [comment]
Fixed in b85e10184efd13b3a7b192de35f80a6ad842ee57.
